tests/pom.xml

```diff
@@ -39,6 +40,34 @@
     <pluginManagement>
       <plugins>
         <plugin>
+          <groupId>com.google.cloud.tools</groupId>
+          <artifactId>appengine-maven-plugin</artifactId>
+          <configuration>
+            <deploy.project>${app.deploy.project}</deploy.project>
+            <deploy.version>${maven.build.timestamp}</deploy.version>

```

Comment:
I don't think you need to specify `deploy.project` and `deploy.version` since you're using system properties `app.deploy.project` and `app.deploy.version` below.